src/wait.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{elements::Element, error, Client, Locator};
+use core::fmt;
+use futures_util::TryFutureExt;
+use std::{
+    error::Error,
+    future::Future,
+    pin::Pin,
+    time::{Duration, Instant},
+};
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(30);
+const DEFAULT_PERIOD: Duration = Duration::from_millis(250);
+
+/// Used for setting up a wait operation on the client.
+#[derive(Debug)]
+pub struct Wait<'c> {
+    client: &'c mut Client,
+    timeout: Option<Duration>,
+    period: Duration,
+}
+
+impl<'c> Wait<'c> {
+    /// Create a new wait operation from a client.
+    pub fn new(client: &'c mut Client) -> Self {
+        Self {
+            client,
+            timeout: Some(DEFAULT_TIMEOUT),
+            period: DEFAULT_PERIOD,
+        }
+    }
+
+    /// Set the timeout until the operation should wait.
+    pub fn until(mut self, timeout: Duration) -> Self {
```

Comment:
How about the name `at_most` instead? `until` implies it takes an `Instant`/`SystemTime` to me at least, not a `Duration`.